scout/supporting_data/thermal_loads_data/resstock_to_thermalLoads.py

```diff
@@ -0,0 +1,240 @@
+                keep_cols = [
+                    c
+                    for c in subset.columns
+                    if (
+                        "component_load_" + euse_lower in str(c).lower()
+                        or c in ["CDIV", "BLDG", "NBLDGS"]
+                    )
+                ]
+                subset = subset[keep_cols]
+
+                # Compute weights and prepare a category accumulator
+                sum_bldgs = subset["NBLDGS"].sum()
+                # Per-row weight; if sum is zero, set weight to 0 (vector of zeros via broadcasting)
+                row_weight = subset["NBLDGS"] / sum_bldgs if sum_bldgs > 0 else 0.0
+
+                # Accumulate weighted totals per category in this dict
+                cat_weighted = {cat: 0.0 for cat in categories}
+                # PEOPLE is explicitly zero per your comment
+                cat_weighted["PEOPLE"] = 0.0
+
+                for segment, category in RESSTOCK_SEGMENT_TO_CATEGORY.items():
+                    seg_cols = [
+                        col
+                        for col in subset.columns
+                        if ("component_load_" + euse_lower in str(col).lower())
+                        and (segment in str(col).lower())
+                    ]
+
+                    # Row-wise sum across all matched segment columns
+                    seg_row_sum = subset[seg_cols].sum(axis=1)
+                    seg_weighted_value = float((seg_row_sum * row_weight).sum())
+                    cat_weighted[category] += seg_weighted_value
+
+                final_data = pd.concat(
+                    [
+                        final_data,
+                        pd.DataFrame(
+                            {
+                                "CDIV": cdiv,
+                                "BLDG": bldg,
+                                "ENDUSE": output_enduse,
+                                "NBLDGS": sum_bldgs,
+                            },
+                            index=[0],
+                        ),
+                    ],
+                    ignore_index=True,
+                )
+
+                # weighted thermal components (compute scalar weighted averages per row)
+                row_mask = (
+                    (final_data["CDIV"] == cdiv)
+                    & (final_data["BLDG"] == bldg)
+                    & (final_data["ENDUSE"] == output_enduse)
+                )
+
+                for cat, val in cat_weighted.items():
+                    final_data.loc[row_mask, cat] = val
+
+                # Normalize to shares (fractions of total)
+                share_components = [c for c in categories if c != "TOTAL"]
+
+                # Sum across share components for just this row (Series of length 1)
+                row_total = final_data.loc[row_mask, share_components].sum(axis=1)
+
+                # Normalize only if positive (avoid divide-by-zero)
+                if not row_total.empty and float(row_total.iloc[0]) > 0:
```

Comment:
I don't think we would ever get negative total load at this geography, but I think technically it should be:

```suggestion
                if not row_total.empty and float(row_total.iloc[0]) != 0:
```